docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<img src = "images/ArchitectureDiagram.png">
 
+The above **Architecture Diagram** provides the high-level design of the application.
+
+### Main components
+
+The application comprises the following components:
+
+* `Main`, which handles program startup and shutdown, and also interactions between other components.
+* `UI`, which handles user input and showing messages to the user.
+* `Storage`, which handles the loading and saving of data upon program startup and shutdown.
+* `Parser`, which converts user input into commands.
+* `Command`, which are executed to modify the data stored in the program.
+* `Event`, which stores the program's data.
+
+### Interactions between components
+
+<img src = "images/ArchitectureSequenceDiagram.png">
+
+The above **Sequence Diagram** shows how the different components of the system interact with one
+another in the scenario when the command `add -e event -t 1200 -v venue` is executed.
+
+### UI component
+
+<img src = "images/UiClassDiagram.png">
+
+The `UI` component comprises an Ui class, as shown in the above **Class Diagram**.
+
+The `UI` does the following:
+
+* Take in command input from the user and pass the input to `Main`.
+* Show output messages from `Command` to the user after command execution.
+
+### Storage component
+
+<img src = "images/StorageClassDiagram.png">
+
+The `Storage` component comprises a Storage class, as shown in above **Class Diagram**.
+
+The `Storage` does the following:
+
+* Load events and participants information from a text file and save it to `EventList` list.
+* Save events from `EventList` list to a text file
+
+### Command component
+
+The `Command` component and its component classes are shown in the below **Class Diagram**:
+
+<img src = "images/CommandClassDiagram.png">
+
+The `Command` component does the following:
+
+* Handle the execution of the user command through interactions with `Event`.
+* Provides a command output message to `Ui` after the command execution.
+
+The interactions between `Command` and other commands in the system is shown in the following _Sequence Diagram_:
+
+<img src = "images/CommandSequenceDiagram.png">
```

Comment:
The diagram only describes the interaction between a single `XYZCommand` and the other components of the system, yet the text description above does not match the diagram as it says that it is showing interactions between various commands. Consider proofreading content before updating?

Reply:
Thanks for catching this, was an oversight on our part (it was meant to be "components" rather than "commands")